PaulasWeatherApp/PaulasWeatherApp/ViewController/CurrentWeatherViewController.swift

```diff
@@ -8,11 +8,108 @@
     }
     
+    private func setActivityIndicator() {
+        activityIndicator.center = view.center
+        activityIndicator.hidesWhenStopped = true
+        view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
+    }
+    
+    private func setTempLabel() {
+        
+        tempLabel.translatesAutoresizingMaskIntoConstraints = false
+        tempLabel.textAlignment = .center
+        tempLabel.font = .systemFont(ofSize: 36)
+        tempLabel.textColor = .black
+        tempLabel.text = "... °C"
+        tempLabel.backgroundColor = .clear
+        tempLabel.sizeToFit()
+        
+        view.addSubview(tempLabel)
+        
+        NSLayoutConstraint.activate([
+            tempLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            tempLabel.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -70),
+        ])
+    }
+    
+    private func setSunsetLabel() {
+        sunsetLabel.translatesAutoresizingMaskIntoConstraints = false
+        sunsetLabel.textAlignment = .center
+        sunsetLabel.font = .systemFont(ofSize: 12)
+        sunsetLabel.textColor = .black
+        sunsetLabel.backgroundColor = .clear
+        sunsetLabel.attributedText = sunsetString
+        sunsetLabel.sizeToFit()
+        
+        view.addSubview(sunsetLabel)
+        
+        NSLayoutConstraint.activate([
+            sunsetLabel.leadingAnchor.constraint(equalTo: sunriseLabel.trailingAnchor,constant: 10),
+            sunsetLabel.trailingAnchor.constraint(equalTo: tempLabel.trailingAnchor),
+            sunsetLabel.topAnchor.constraint(equalTo: tempLabel.bottomAnchor)
+        ])
+    }
+    
+    private func setSunriseLabel() {
+        sunriseLabel.translatesAutoresizingMaskIntoConstraints = false
+        sunriseLabel.textAlignment = .center
+        sunriseLabel.font = .systemFont(ofSize: 12)
+        sunriseLabel.textColor = .black
+        sunriseLabel.backgroundColor = .clear
+        sunriseLabel.attributedText = sunriseString
+        sunriseLabel.sizeToFit()
+        view.addSubview(sunriseLabel)
+        
+        NSLayoutConstraint.activate([
+            sunriseLabel.leadingAnchor.constraint(equalTo: tempLabel.leadingAnchor),
+            sunriseLabel.topAnchor.constraint(equalTo: tempLabel.bottomAnchor)
+        ])
+    }
+    
+     func setCurrentWeatherData(_ data : WeatherDataModel) {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "hh:mm a"
+        
+        tempLabel.text = "\(data.current.temp)  °C"
+        guard let sunset = data.current.sunset, let sunrise = data.current.sunrise  else { return }
+        
+        sunsetString.append(NSAttributedString(string: dateFormatter.string(from: Date(timeIntervalSince1970: Double(sunset))) ))
+        sunsetLabel.attributedText = sunsetString
+        
+        sunriseString.append(NSAttributedString(string: dateFormatter.string(from: Date(timeIntervalSince1970:Double(sunrise)))))
```

Comment:
Minor: a space is needed before Double(sunrise)
I'm leaving all these comments because a lot of apps have coding tools installed, and if you don't follow the "rules" that are agreed for the project you'll get an error. For example in GCM we have Swift lint and it gets very annoying sometimes. 
You can read more about this tool here https://github.com/realm/SwiftLint